libs/core/langchain_core/indexing/api.py

```diff
@@ -265,6 +268,9 @@ def index(
             "delete" and "add_documents" required methods.
         ValueError: If source_id_key is not None, but is not a string or callable.
     """
+    # Local scope to avoid circular imports
+    from langchain_core.vectorstores import VectorStore
```

Comment:
As a rule of thumb, circular imports tend to cause subtle problems are are usually best avoided all together.